Assets/Scripts/CardIndexController.cs

```diff
@@ -27,12 +33,16 @@ public class CardIndexController : MonoBehaviour {
         }
     }
 
-    public void SetCanCast (bool canCast) {
+    void SetIsDisplayed (bool isDisplayed) {
+        this.isDisplayed = isDisplayed;
+    }
+
+    void SetCanCast (bool canCast) {
```

Comment:
Why does `canCast` need to be a member variable if it's only used in `UpdateColor()`? Same with `isDisplayed()`.

Instead of having `UpdateIndexDisplay()` set all of these variables then changes the views with the updated member variables, why not just have `UpdateIndexDisplay()` call `UpdateIndex()`, `UpdateText()`, and `UpdateColor()` with its arguments?